src/algorithms/neal2_algorithm.cpp

```diff
@@ -6,11 +6,60 @@
 #include <vector>
 
 #include "../../proto/cpp/marginal_state.pb.h"
+#include "../collectors/base_collector.hpp"
 #include "../hierarchies/base_hierarchy.hpp"
 #include "../mixings/base_mixing.hpp"
 #include "../utils/distributions.hpp"
 #include "../utils/rng.hpp"
 
+//! \param grid Grid of points in matrix form to evaluate the density on
+//! \param coll Collector containing the algorithm chain
+//! \return     Matrix whose i-th column is the lpdf at i-th iteration
+Eigen::MatrixXd Neal2Algorithm::eval_lpdf(
```

Comment:
quick question: why is this moved here from MarginalAlgorithm?

Reply:
Because there it would also be used by the dependent algorithms.

But, now that I think about it, we can just override the method for dependent algorithms :sweat_smile: I'll go with that, it didn't come to mind but it definitely makes more sense